includes/Order/Ajax.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace WeDevs\Dokan\Order;
+
+// don't call the file directly
+if ( ! defined( 'ABSPATH' ) ) {
+    exit;
+}
+
+/**
+ * Class Ajax
+ *
+ * @since DOKAN_SINCE
+ *
+ * @package WeDevs\Dokan\Order
+ */
+class Ajax {
+    /**
+     * Class constructor
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return void
+     */
+    public function __construct() {
+        add_action( 'wp_ajax_dokan_search_downloadable_products', [ $this, 'search_downloadable_products' ] );
+    }
+
+    /**
+     * Search downloadable products
+     *
+     * @since DOKAN_SINCE
+     *
+     * @return void
+     */
+    public function search_downloadable_products() {
+        check_ajax_referer( 'search-downloadable-products', '_nonce' );
+
+        $limit = absint( apply_filters( 'woocommerce_json_search_limit', 30 ) );
+        if ( ! empty( $_GET['limit'] ) ) {
+            $limit = absint( $_GET['limit'] );
+        }
+
+        $page = 1;
+        if ( ! empty( $_GET['page'] ) ) {
+            $page = absint( $_GET['page'] );
+        }
+
+        $term = isset( $_GET['term'] ) ? (string) wc_clean( wp_unslash( $_GET['term'] ) ) : '';
+        $args = [
+            'status'       => 'publish',
+            'downloadable' => true,
+            'author'       => get_current_user_id(),
```

Comment:
We should consider Support for vendor Staff here. Please replace `get_current_user_id()` with `dokan_get_current_user_id()`.